org.eclipse.winery.workflowmodeler/src/app/services/winery.service.ts

```diff
@@ -0,0 +1,158 @@
+        this.repositoryURL = queryParams.repositoryURL;
+        this.namespace = queryParams.namespace;
+        this.serviceTemplateId = queryParams.id;
+        this.plan = queryParams.plan;
+
+        if (!isNullOrUndefined(this.repositoryURL)) {
+            this.loadPlan();
+        }
+
+    }
+
+    public loadNodeTemplates(): Observable<any> {
+        const url = 'servicetemplates/' + this.encode(this.namespace)
+            + '/' + this.encode(this.serviceTemplateId) + '/topologytemplate/';
+
+        return this.httpService.get(this.getFullUrl(url)).map(response => {
+            const nodeTemplates = [];
+            for (const key in response.nodeTemplates) {
+                if (response.nodeTemplates.hasOwnProperty(key)) {
+                    const nodeTemplate = response.nodeTemplates[key];
+                    nodeTemplates.push({
+                        id: nodeTemplate.id,
+                        type: nodeTemplate.type.replace(/^\{(.+)\}(.+)/, '$2'),
+                        name: nodeTemplate.name,
+                        namespace: nodeTemplate.type.replace(/^\{(.+)\}(.+)/, '$1'),
+                    });
+                }
+            }
+            return nodeTemplates;
+        });
+    }
+
+    public loadTopologyProperties(nodeTemplate: NodeTemplate) {
+        const url = 'nodetypes/' + this.encode(nodeTemplate.namespace)
+            + '/' + this.encode(nodeTemplate.id) + '/propertiesdefinition/winery/list/';
+
+        this.httpService.get(this.getFullUrl(url)).subscribe(properties => {
+            properties.forEach(property => nodeTemplate.properties.push(property.key));
+        });
+    }
+
+    public loadNodeTemplateInterfaces(namespace: string, nodeType: string): Observable<any> {
+        const url = 'nodetypes/' + this.encode(namespace)
+            + '/' + this.encode(nodeType) + '/interfaces/';
+
+        return this.httpService.get(this.getFullUrl(url));
+    }
+
+    public loadNodeTemplateOperations(namespace: string,
+                                      nodeType: string,
+                                      interfaceName: string): Observable<any> {
+        const url = 'nodetypes/' + this.encode(namespace)
+            + '/' + this.encode(nodeType) + '/interfaces/' + this.encode(interfaceName) + '/operations/';
+
+        return this.httpService.get(this.getFullUrl(url));
+    }
+
+    public loadNodeTemplateOperationParameter(namespace: string,
+                                              nodeType: string,
+                                              interfaceName: string,
+                                              operation: string): Promise<any> {
+        const relativePath = 'nodetypes/' + this.encode(namespace) + '/' + this.encode(nodeType)
+            + '/interfaces/' + this.encode(interfaceName) + '/operations/' + this.encode(operation) + '/';
+
+        // input parameters
+        let inputPromise: Promise<any> = this.httpService

```

Comment:
You don't reassign the variable here, so you could use `const` instead? The promise type should also be known, I think.
Same applies for the `outputPromise` bleow.